dpnp/dpnp_iface_manipulation.py

```diff
@@ -1954,6 +1943,105 @@ def repeat(a, repeats, axis=None):
     return dpnp_array._create_from_usm_ndarray(usm_res)
 
 
+def require(a, dtype=None, requirements=None, *, like=None):
+    """
+    Return an dpnp.ndarray of the provided type that satisfies requirements.
+
+    This function is useful to be sure that an array with the correct flags
+    is returned for passing to compiled code (perhaps through ctypes).
+
+    For full documentation refer to :obj:`numpy.require`.
+
+    Parameters
+    ----------
+    a : array_like
+       The object to be converted to a type-and-requirement-satisfying array.
+    dtype : data-type, optional
+       The required data-type. If None preserve the current dtype. If your
+       application requires the data to be in native byteorder, include
+       a byteorder specification as a part of the dtype specification.
+    requirements : {str, sequence of str}, , optional
+       The requirements list can be any of the following:
+
+       * 'F_CONTIGUOUS' ('F') - ensure a Fortran-contiguous array
+       * 'C_CONTIGUOUS' ('C') - ensure a C-contiguous array
+       * 'WRITABLE' ('W') - ensure a writable array
+
+    Returns
+    -------
+    out : dpnp.ndarray
+        Array with specified requirements and type if given.
+
+    Limitations
+    -----------
+    Parameter `like` is supported only with default value ``None``.
+    Otherwise, the function raises `NotImplementedError` exception.
+
+    See Also
+    --------
+    :obj:`dpnp.asarray` : Convert input to an ndarray.
+    :obj:`dpnp.asanyarray ` : Convert to an ndarray, but pass through
+                        ndarray subclasses.
+    :obj:`dpnp.ascontiguousarray` : Convert input to a contiguous array.
+    :obj:`dpnp.asfortranarray` : Convert input to an ndarray with
+                        column-major memory order.
+    :obj:`dpnp.ndarray.flags` : Information about the memory layout
+                        of the array.
+
+    Notes
+    -----
+    The returned array will be guaranteed to have the listed requirements
+    by making a copy if needed.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> x = np.arange(6).reshape(2,3)
+    >>> x.flags
+      C_CONTIGUOUS : True
+      F_CONTIGUOUS : False
+      WRITEABLE : True
+
+    >>> y = np.require(x, dtype=np.float32, requirements=['W', 'F'])
+    >>> y.flags
+      C_CONTIGUOUS : False
+      F_CONTIGUOUS : True
+      WRITEABLE : True
+
+    """
+
+    dpnp.check_limitations(like=like)
+
+    possible_flags = {
+        "C": "C",
+        "C_CONTIGUOUS": "C",
```

Comment:
What about support of `'CONTIGUOUS': 'C'` and `'FORTRAN': 'F'` as present in numpy and cupy?

Reply:
We can add `'FORTRAN'` but we generally do not have it in `dpnp` (and `dpctl` ), so I did not.

```python
>>> import dpnp, numpy
>>> a_np = numpy.ones((10, 5))
>>> a_dp = dpnp.ones((10, 5))
>>> a_dp.flags["FORTRAN"] # returns None

>>> a_np.flags["FORTRAN"]
False
```

And for `'CONTIGUOUS'`, NumPy consider it equivalent to `'C_CONTIGUOUS'`  which is different than the behavior that we have in `dpnp` (and `dpctl`) and behavior in `dpnp` make more sense to me. Both `'C_CONTIGUOUS'` and `'F_CONTIGUOUS'` are `'CONTIGUOUS'`.

```python
>>> import dpnp, numpy
>>> a_np = numpy.ones((10,5), order="F")
>>> a_dp = dpnp.ones((10,5), order="F")
>>> a_dp.flags["CONTIGUOUS"]
True
>>> a_np.flags["CONTIGUOUS"]
False
```

Reply:
I see, it's clear now. Thank you.
I tend to agree that they are not require for dpnp here.